File: numberphile/switch/switch_py/switch.py
#!/usr/bin/env python3

def print_grid(person, buttons):
    n = len(buttons) + 1
    print("person: "+str(person))
    for button in range(1,n):
        print(buttons[button-1],end='')
        if (button % 10 == 0):
            print()
    print()

def switch_logic(person, buttons):
    n = len(buttons) + 1
    for button in range(1, n):
        if ( button % person == 0 ):
            buttons[button-1] ^= 1

if __name__ == "__main__":
    n = 100
    buttons=[0] * n

    for person in range(1, n + 1):
        switch_logic(person, buttons)
        print_grid(person, buttons)

File: numberphile/switch/switch_c/switch.c
#include <stdlib.h>
#include <stdio.h>

void print_grid(const size_t person, const char *buttons, const size_t size){
    size_t n = size + 1;
    printf("person: %ld\n", person);
    for(int i=1; i<n; i++){
        printf("%d", buttons[i-1]);
        if (i % 10 == 0) printf("\n");
    }
    printf("\n");
}

void switch_logic(const size_t person, char *buttons, const size_t size){    
    for (int button=1; button < size+1; button++){
        if (button % person == 0) buttons[button-1] ^= 1;
    }
}

void fondle_my_knobs(const size_t n, char * buttons){
    for(int person=1; person < n+1; person++){
        switch_logic(person, buttons, n);
        print_grid(person, buttons, n);    
    }
}

int main(){
    size_t n = 100; 
    
    char *buttons = calloc(n,1);

    fondle_my_knobs(n, buttons);
    
    free(buttons);
    return 0;
}
